baseboard_windows.go

```diff
@@ -0,0 +1,32 @@
+// Use and distribution licensed under the Apache license version 2.
+//
+// See the COPYING file in the root project directory for full text.
+//
+
+package ghw
+
+import "github.com/StackExchange/wmi"
+
+type Win32_BaseBoard struct {
+	Manufacturer string
+	SerialNumber string
+	Tag          string
+	Version      string
+}
+
+func (ctx *context) baseboardFillInfo(info *BaseboardInfo) error {
+	// Getting disk drives from WMI
+	var win32BaseboardDescriptions []Win32_BaseBoard
+	q1 := wmi.CreateQuery(&win32BaseboardDescriptions, "")
```

Comment:
You don't need to use the `wmi.CreateQuery()` function here. Instead, simply create a const string with the WQL query, like so:

```go
const (
        wqlBaseboard = "SELECT Manufacturer, SerialNumber, Tag, Version FROM Win32_BaseBoard"
)

type win32Baseboard struct {
	Manufacturer string
	SerialNumber string
	Tag          string
	Version      string
}

func (ctx *context) baseboardFillInfo(info *BaseboardInfo) error {
        var win32BaseboardDescriptions []win32Baseboard
        if err := wmi.Query(wqlBaseboard, &win32BaseboardDescriptions); err != nil {
		return err
	}
	if len(win32BaseboardDescriptions) > 0 {
		info.AssetTag = win32BaseboardDescriptions[0].Manufacturer
		info.SerialNumber = win32BaseboardDescriptions[0].Manufacturer
		info.Vendor = win32BaseboardDescriptions[0].Manufacturer
		info.Version = win32BaseboardDescriptions[0].Manufacturer
	}
	return nil
}
```

Reply:
That is true.

The main reason I was using the same pattern was because I could just change the **win32Baseboard** struct obtaining dynamically the correct updated query.

Pretty usefull if you are lazy like me and you are always exploring WMI tables to find the correct information ^^

Let me know if you still think it's better to use a static query instead of having a dynamic one built from the structure.

Reply:
Yeah, I think I'd prefer to just use a static const string query versus using an exported/public struct named the same as the WQL database table. It's just cleaner IMHO :)